Make pytest.mark.asyncio compatible with classic functions

Tadaboody opened the pull request:
Closes #109

thilp commented on tests/test_simple_35.py:
```diff
@@ -72,6 +72,10 @@ def test_unused_port_factory_fixture(unused_tcp_port_factory, event_loop):
 
+@pytest.mark.asyncio
+def test_non_corutine_marked():
+    """Tests that marked non-corutines function normally (Issue #109)"""
```
```suggestion
    """Tests that marked non-coroutines function normally (Issue #109)"""
```

blueyed commented:
LGTM, please rebase and take the suggested doc fix.

mitzkia commented:
@Tadaboody could you please update this PR (rebase and doc update). It would help me a lot.

Tadaboody commented:
Sorry this took so long 😅 wasn't really maintaining my github

Tadaboody commented:
It looks like currently there isn't a problem with non-coroutines I deleted my old change (it had problems with hypothesis) and the test I added still passed. Is it enough to say that the issue is resolved?
